src/shared/install.c

```diff
@@ -603,6 +604,11 @@ static int remove_marked_symlinks_fd(
                         if (!found)
                                 continue;
 
+                        if (dry_run) {
+                                unit_file_changes_add(changes, n_changes, UNIT_FILE_UNLINK, p, NULL);
+                                continue;
+                        }
```

Comment:
well, but it will return incorrect information, since it will not follow the symlinks anymore...

The current logic is to to set the `*restart` bool whenever there's reason to iterate the directory tree again. As it appears this bool should not be set in some of these cases, so that we don't restart over and over again, just because we see the same link over and over again.